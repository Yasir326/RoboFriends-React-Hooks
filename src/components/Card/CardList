// import React from 'react';
// import { shallow } from 'enzyme';
// import { Provider } from 'react-redux';
// import configureStore from 'redux-mock-store'
// import { CardList } from './CardList';
// import '../../setUpTest';

// const mockRobots = [
//   {
//     id: 1,
//     name: 'robocop',
//     email: 'robocop@gmail.com',
//     key: 1,
//   },
// ];

// const mockStore = configureStore()
// let store

// it('expect to render Card component', () => {
//   store = mockStore(mockRobots)
//   // const cardlist = shallow(<Provider<CardList/></Provider>, mockRobots);
//   expect(
//       <Provider>
//         <CardList store={store}/>
//       </Provider>
//   ).toMatchSnapshot();
// });
