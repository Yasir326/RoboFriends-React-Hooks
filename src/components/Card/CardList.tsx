import { useSelector } from 'react-redux';
import React from 'react';
import Card from './Card';
import { IRobots } from '../../types';



const CardList = () => {
  const robots: Array<IRobots> = useSelector((state) =>
    state.requestRobots.robots.filter(
      (robot: { name: string }) => {
        return robot.name
          .toLowerCase()
          .includes(state.searchRobots.searchField.toLowerCase());
      },
      {
        error: state.requestRobots.pending,
      }
    )
  );

  return (
    <div>
      {robots.map((robot) => {
        return (
          <Card
            id={robot.id}
            name={robot.name}
            email={robot.email}
            key={robot.id}
          />
        );
      })}
      ;
    </div>
  );
};

export default CardList;
