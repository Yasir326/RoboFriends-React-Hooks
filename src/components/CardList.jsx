import { useSelector } from 'react-redux';
import React from 'react';
import Card from './Card';

const CardList = () => {
  const robots = useSelector((state) =>
    state.requestRobots.robots.filter(
      (robot) => {
        return robot.name
          .toLowerCase()
          .includes(state.searchRobots.searchField.toLowerCase());
      },
      {
        error: state.requestRobots.pending,
      }
    )
  );

  return (
    <div>
      {robots.map((robot) => {
        return (
          <Card
            id={robot.id}
            name={robot.name}
            email={robot.email}
            key={robot.id}
          />
        );
      })}
      ;
    </div>
  );
};

export default CardList;
