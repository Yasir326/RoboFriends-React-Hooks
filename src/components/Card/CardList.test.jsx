import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import CardList from './CardList';
import '../../setUpTest';

let store;
let wrapper;

beforeEach(() => {
  const mockStore = configureStore();
  const initialState = {
    robots: [
      {
        id: 1,
        name: 'robocop',
        email: 'robocop@gmail.com',
        key: 1,
      },
      {
        id: 2,
        name: 'bart',
        email: 'bartp@gmail.com',
        key: 2,
      },
    ],
    error: false,
  };

  store = mockStore(initialState);
  wrapper = shallow(
    <Provider store={store}>
      <CardList {...initialState} />
    </Provider>
  );
});

it('expect to render Card component', () => {
  expect(wrapper).toMatchSnapshot();
});
