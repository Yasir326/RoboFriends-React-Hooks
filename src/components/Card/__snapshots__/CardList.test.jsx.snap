// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CardList>
    <Card
      robots={
        Array [
          Object {
            "email": "robocop@gmail.com",
            "id": 1,
            "key": 1,
            "name": "robocop",
          },
        ]
      }
    />
  </CardList>
</Provider>
`;
